lib/screen/chat/chat_thread_page.dart

```diff
@@ -0,0 +1,420 @@
+import 'dart:convert';
+
+import 'package:autonomy_flutter/common/environment.dart';
+import 'package:autonomy_flutter/model/chat_message.dart';
+import 'package:autonomy_flutter/util/constants.dart';
+import 'package:autonomy_flutter/util/datetime_ext.dart';
+import 'package:autonomy_flutter/util/log.dart';
+import 'package:autonomy_flutter/util/string_ext.dart';
+import 'package:autonomy_flutter/util/style.dart';
+import 'package:autonomy_flutter/view/responsive.dart';
+import 'package:autonomy_theme/autonomy_theme.dart';
+import 'package:crypto/crypto.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:eth_sig_util/util/utils.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_chat_types/flutter_chat_types.dart' as types;
+import 'package:autonomy_flutter/model/chat_message.dart' as app;
+import 'package:autonomy_flutter/view/back_appbar.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_chat_ui/flutter_chat_ui.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:libauk_dart/libauk_dart.dart';
+import 'package:uuid/uuid.dart';
+import 'package:web_socket_channel/io.dart';
+import 'package:web_socket_channel/web_socket_channel.dart';
+
+class ChatThreadPage extends StatefulWidget {
+  static const String tag = "chat_thread_page";
+  final ChatThreadPagePayload payload;
+
+  const ChatThreadPage({Key? key, required this.payload}) : super(key: key);
+
+  @override
+  State<ChatThreadPage> createState() => _ChatThreadPageState();
+}
+
+class _ChatThreadPageState extends State<ChatThreadPage> {
+  final List<types.Message> messages = [];
+  late types.User user;
+  WebSocketChannel? _websocketChannel;
+  late ChatThreadPagePayload payload;
+  int? lastMessageTimestamp;
+  bool didFetchAllMessages = false;
+  String? historyRequestId;
+  bool isTyping = false;
+  bool stopConnect = false;
```

Comment:
@phuocbitmark can you please make all the fields private if it's not necessary to be public?